File: src/main/java/com/matag/game/turn/action/counters/CountersService.java
package com.matag.game.turn.action.counters;

import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.stereotype.Component;

import com.matag.cards.ability.type.AbilityType;
import com.matag.game.cardinstance.CardInstance;

@Component
public class CountersService {
  private static final Logger LOGGER = LoggerFactory.getLogger(CountersService.class);

  public void addPlus1Counters(CardInstance target, int counters) {
    if (counters > 0) {
      target.getModifiers().getCounters().addPlus1Counters(1);
      LOGGER.info(target.getIdAndName() + " got " + counters + " +1/+1 counters");
    }
  }

  public void addMinus1Counters(CardInstance target, int counters) {
    if (counters > 0) {
      target.getModifiers().getCounters().addMinus1Counters(1);
      LOGGER.info(target.getIdAndName() + " got " + counters + " -1/-1 counters");
    }
  }

  public void addKeywordCounter(CardInstance target, AbilityType keywordCounter) {
    target.getModifiers().getCounters().addKeywordCounter(keywordCounter);
    LOGGER.info(target.getIdAndName() + " got " + keywordCounter + " counters");
  }
}

File: src/main/kotlin/com/matag/game/turn/action/counters/CountersService.kt
package com.matag.game.turn.action.counters

import com.matag.cards.ability.type.AbilityType
import com.matag.game.cardinstance.CardInstance
import org.slf4j.Logger
import org.slf4j.LoggerFactory
import org.springframework.stereotype.Component

@Component
class CountersService {
    fun addPlus1Counters(target: CardInstance, counters: Int) {
        if (counters > 0) {
            target.modifiers.counters.addPlus1Counters(1)
            LOGGER.info(target.idAndName + " got " + counters + " +1/+1 counters")
        }
    }

    fun addMinus1Counters(target: CardInstance, counters: Int) {
        if (counters > 0) {
            target.modifiers.counters.addMinus1Counters(1)
            LOGGER.info(target.idAndName + " got " + counters + " -1/-1 counters")
        }
    }

    fun addKeywordCounter(target: CardInstance, keywordCounter: AbilityType) {
        target.modifiers.counters.addKeywordCounter(keywordCounter)
        LOGGER.info(target.idAndName + " got " + keywordCounter + " counters")
    }

    companion object {
        private val LOGGER: Logger = LoggerFactory.getLogger(CountersService::class.java)
    }
}
